acceptance: Unexpected RPC errors at SQL level

The following tests appear to have failed:

[#581914](https://teamcity.cockroachdb.com/viewLog.html?buildId=581914):


```
--- FAIL: acceptance/TestNodeRestart (9.160s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestNodeRestart589298330
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestNodeRestart: TestNodeRestart/runMode=local (9.150s)
zchaos_test.go:152: pq: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
------- Stdout: -------
CockroachDB node starting at 2018-03-21 23:58:07.805884414 +0000 UTC (took 0.6s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:34157
sql:                 postgresql://root@127.0.0.1:38053?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/cockroach-temp274565497
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1
status:              initialized new cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              1
CockroachDB node starting at 2018-03-21 23:58:08.305721678 +0000 UTC (took 0.2s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:36027
sql:                 postgresql://root@127.0.0.1:33997?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/cockroach-temp772722404
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2
status:              initialized new node, joined pre-existing cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              2
CockroachDB node starting at 2018-03-21 23:58:08.313233017 +0000 UTC (took 0.2s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:44283
sql:                 postgresql://root@127.0.0.1:36449?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/3/cockroach-temp533116656
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/3
status:              initialized new node, joined pre-existing cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              3
CockroachDB node starting at 2018-03-21 23:58:08.323926473 +0000 UTC (took 0.2s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:33003
sql:                 postgresql://root@127.0.0.1:44139?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/4
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/4/cockroach-temp690861043
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/4/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/4
status:              initialized new node, joined pre-existing cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              4
CockroachDB node starting at 2018-03-21 23:58:10.966467045 +0000 UTC (took 0.7s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:33065
sql:                 postgresql://root@127.0.0.1:35833?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/cockroach-temp630593282
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2
status:              restarted pre-existing node
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              2
E180321 23:58:15.236541 4802 acceptance/zchaos_test.go:272  round 1: failed to do consistency check against node 3: Consistency checking is unimplmented and should be re-implemented using SQL
CockroachDB node starting at 2018-03-21 23:58:15.939592801 +0000 UTC (took 0.6s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:40267
sql:                 postgresql://root@127.0.0.1:41409?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/cockroach-temp659496049
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1
status:              restarted pre-existing node
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              1
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestNodeRestart589298330
--- FAIL: acceptance/TestNodeRestart (9.160s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestNodeRestart589298330
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestNodeRestart: TestNodeRestart/runMode=local (9.150s)
zchaos_test.go:152: pq: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
------- Stdout: -------
CockroachDB node starting at 2018-03-21 23:58:07.805884414 +0000 UTC (took 0.6s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:34157
sql:                 postgresql://root@127.0.0.1:38053?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/cockroach-temp274565497
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1
status:              initialized new cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              1
CockroachDB node starting at 2018-03-21 23:58:08.305721678 +0000 UTC (took 0.2s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:36027
sql:                 postgresql://root@127.0.0.1:33997?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/cockroach-temp772722404
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2
status:              initialized new node, joined pre-existing cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              2
CockroachDB node starting at 2018-03-21 23:58:08.313233017 +0000 UTC (took 0.2s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:44283
sql:                 postgresql://root@127.0.0.1:36449?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/3/cockroach-temp533116656
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/3
status:              initialized new node, joined pre-existing cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              3
CockroachDB node starting at 2018-03-21 23:58:08.323926473 +0000 UTC (took 0.2s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:33003
sql:                 postgresql://root@127.0.0.1:44139?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/4
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/4/cockroach-temp690861043
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/4/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/4
status:              initialized new node, joined pre-existing cluster
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              4
CockroachDB node starting at 2018-03-21 23:58:10.966467045 +0000 UTC (took 0.7s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:33065
sql:                 postgresql://root@127.0.0.1:35833?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/cockroach-temp630593282
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/2
status:              restarted pre-existing node
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              2
E180321 23:58:15.236541 4802 acceptance/zchaos_test.go:272  round 1: failed to do consistency check against node 3: Consistency checking is unimplmented and should be re-implemented using SQL
CockroachDB node starting at 2018-03-21 23:58:15.939592801 +0000 UTC (took 0.6s)
build:               CCL v2.0-alpha.20180212-1841-g5029c38 @ 2018/03/21 23:43:14 (go1.10)
admin:               http://127.0.0.1:40267
sql:                 postgresql://root@127.0.0.1:41409?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestNodeRestart/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/cockroach-temp659496049
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster214252337/1
status:              restarted pre-existing node
clusterID:           41aeb9fb-64c5-4bc0-b260-ad06c783f409
nodeID:              1
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestNodeRestart589298330

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/testutils/error.go
+++ b/pkg/testutils/error.go
@@ -60,10 +60,16 @@ func IsPError(pErr *roachpb.Error, re string) bool {
 // itself. This can occur when a node is restarting or is unstable in
 // some other way. Note that retryable errors may occur event in cases
 // where the SQL execution ran to completion.
+//
+// TODO(bdarnell): Why are RPC errors in this list? These should
+// generally be retried on the server side or transformed into
+// ambiguous result errors ("connection reset/refused" are needed for
+// the pgwire connection, but anything RPC-related should be handled
+// within the cluster).
 func IsSQLRetryableError(err error) bool {
 	// Don't forget to update the corresponding test when making adjustments
 	// here.
-	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF|result is ambiguous)")
+	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|rpc error: code = Unavailable|EOF|result is ambiguous)")
 }
 
 // Caller returns filename and line number info for the specified stack

